src/core/server/legacy/plugins/get_nav_links.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { LegacyUiExports, LegacyNavLink, LegacyPluginSpec } from '../types';
+
+const REMOVE_FROM_ARRAY: LegacyNavLink[] = [];
+
+function getUiAppsNavLinks({ uiAppSpecs = [] }: LegacyUiExports, pluginSpecs: LegacyPluginSpec[]) {
+  return uiAppSpecs.flatMap(spec => {
+    if (!spec) {
+      return REMOVE_FROM_ARRAY;
+    }
+
+    const id = spec.id;
+
+    if (!id) {
+      throw new Error('Every app must specify an id');
+    }
+
+    if (spec.pluginId && !pluginSpecs.some(plugin => plugin.getId() === spec.pluginId)) {
+      throw new Error(`Unknown plugin id "${spec.pluginId}"`);
+    }
+
+    const listed = typeof spec.listed === 'boolean' ? spec.listed : true;
+
+    if (spec.hidden || !listed) {
+      return REMOVE_FROM_ARRAY;
+    }
+
+    return {
+      id,
+      category: spec.category,
+      title: spec.title,
+      order: typeof spec.order === 'number' ? spec.order : 0,
+      icon: spec.icon,
```

Comment:
what about other fields from https://github.com/elastic/kibana/pull/57542/files#diff-d1ea0ed9e7f4c8bacd7b2ecc4453fa05R62-R74 ? `subUrlBase`, `disableSubUrlTracking`, `tooltip`, etc.

Reply:
`disableSubUrlTracking` seems to be indeed missing when compared to the pre-#52161 version, even if not present in our TS type

```ts
export type LegacyAppSpec = Pick<
  ChromeNavLink,
  'title' | 'order' | 'icon' | 'euiIconType' | 'url' | 'linkToLastSubUrl' | 'hidden' | 'category'
> & { pluginId?: string; id?: string; listed?: boolean };
```

However the other properties were just not present when generating the navlink from the legacy app... So I guess I should not add them, wdyt?

https://github.com/elastic/kibana/blob/8c2980213e1ca9e6f7d9f68e22b80148af03c02c/src/legacy/ui/ui_apps/ui_app.js#L67-L77

Reply:
> So I guess I should not add them, wdyt?

agree. probably we should clean up types as well

Reply:
I ended up by doing the opposite of what I suggested to be able to factorize the mapping method...

```ts
export type LegacyNavLinkSpec = Partial<ChromeNavLink>;

export type LegacyAppSpec = Partial<ChromeNavLink> & {
  pluginId?: string;
  listed?: boolean;
};
```

PTAL, tell me if you think this is a risk.

Reply:
https://github.com/elastic/kibana/blob/8c2980213e1ca9e6f7d9f68e22b80148af03c02c/src/legacy/ui/ui_nav_links/__tests__/ui_nav_link.js#L81

The typo in `linkToLastSub` instead of `linkToLastSubUrl` was actually countering the wrong `false` default value in the mapping function...